core/src/server/http/http_response.cpp

```diff
@@ -77,6 +77,11 @@ bool IsBodyForbiddenForStatus(server::http::HttpStatus status) {
          (static_cast<int>(status) >= 100 && static_cast<int>(status) < 200);
 }
 
+void AppendToCharArray(char* data, std::string_view what) {
+  char* append_position = data + strlen(data);
```

Comment:
the `strlen` usage here seems off:
1. `SmallString` isn't null-terminated
2. it makes appending cookies quadratic